### ISSUE TITLE
New rule to organize unit test contents

### ISSUE THREAD

**Issue by daniellmb** [daniellmb/angular-test-patterns#13]
@yanivefraim I've always used comments to help organize the contents of a particular unit test such as:

```javascript
// arrange
// act
// assert
```
or 
```javascript
// given
// when
// then
```

What do you think about adding a [new rule](https://github.com/daniellmb/angular-test-patterns/tree/master/spec/rules) to require similar comments within the test patterns?

**Comment by yanivefraim** [comment]
Sounds cool. Are those patterns always relevant?

**Comment by daniellmb** [comment]
No, not *always*, for example, on some unit tests there may be nothing to arrage but I don't think it's detrimental/distracting to have the comments there even if there is nothing in that part of the test. I find it's a nice extention point when you do need to add code to arrage the unit test context, it's clear where that code should go.

**Comment by yanivefraim** [comment]
OK, I'm with you on this (:

**Comment by daniellmb** [comment]
Do you have a preference between "arrange, act, assert" and "given, when, then"?

**Comment by yanivefraim** [comment]
I use "given, when, then", not sure what is better...

**Comment by daniellmb** [comment]
OK, thanks, I don't have a strong preference. I typically use AAA, but GWT may fit a little better with the Jasmine BDD style.
